ts/TabBar/index.tsx

```diff
@@ -0,0 +1,136 @@
+import {MDCTabBar} from '@material/tab-bar';
+import autobind from 'autobind-decorator';
+import {h} from 'preact';
+import MaterialComponent from '../Base/MaterialComponent';
+
+/*
+ * Default props for tabs
+ */
+const defaultProps = {
+  activeTabIndex: 0
+};
+
+export interface ITabLabelProps {}
+export interface ITabLabelState {}
+export class TabLabel extends MaterialComponent<
+  ITabLabelProps,
+  ITabLabelState
+> {
+  protected componentName = 'tab__text-label';
+  protected mdcProps = [];
+
+  @autobind
+  protected materialDom(props) {
+    return <span {...props}>{props.children}</span>;
+  }
+}
+
+export interface ITabIconProps {}
+export interface ITabIconState {}
+export class TabIcon extends MaterialComponent<ITabIconProps, ITabIconState> {
+  protected componentName = 'tab__icon';
+  protected mdcProps = [];
+
+  @autobind
+  protected materialDom(props) {
+    return (
+      <span className="material-icons" {...props}>
+        {props.children}
+      </span>
+    );
+  }
+}
+
+export interface ITabProps extends JSX.HTMLAttributes {
```

Comment:
No need for `extends JSX.HTMLAttributes` for components that extend `MaterialComponent` they do that by them selves.

Reply:
done